### PR TITLE
Feature #58 - Audit trail report

### LINKED ISSUE #58: Als functioneel beheerder wil ik achteraf kunnen reconstrueren hoe de afhandeling van het vernietigingsproces is verlopen,
zodat achteraf onderzocht kan worden hoe de vernietiging is verlopen.
Het gaat hier enerzijds om het vaststellen welke gegevens allemaal worden vastgelegd (de audittrail) en de toegankelijkheid ervan. 

 In aanvulling op de gegevens uit de verklaring kan worden gedacht aan informatie over: 
- wie op welk moment welke stappen in het vernietigingsproces heeft uitgevoerd
- welke opmerkingen zijn toegevoegd
- Welke notificaties naar andere systemen zijn verzonden 
- ? 

[Thema uit conceptofferte: 2. Proces-verbaal/rapport genereren van vernietiging]

*Clearification*

Aan het VL rapport een audit trail toevoegen wie wat gedaan heeft.
Don't mention person names, just roles and times/dates.
Only for the PDF.

### PR DESCRIPTION
**Pull request by SilviaAmAm**
Fixes #58

### CODE REVIEW

**Comment by SilviaAmAm** [comment]
An example of the PDF report with the Audit trail:
Screenshot from 2021-04-08 10-45-42

File: src/archiefvernietigingscomponent/report/templates/report/audittrail/audittrail.html [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        }
+
+        .log-item > .log-item__body {
+            padding-left: 1em;
+        }
+
+    </style>
+</head>
+
+<h2>Audit trail</h2>
```
Comment (annashamray) [inline src/archiefvernietigingscomponent/report/templates/report/audittrail/audittrail.html:28]: I think the "Audit trail" header should be formatted in the same way that a table header above (or be a little smaller, but not bigger)

**Review by annashamray — CHANGES_REQUESTED** [review]
Nice implementation with reusing audit trails! 

The only thing which is bothering me - zaak identifications are not specified in most log entries (creating, updating and reviewing the destruction list).
For example, a record manager created a DL with two zaken A and B
The reviewer asks to remove the zaak A, and the record manager excludes it. 
If they have not specified the identificatie of the zaak A in their comments, archivaris will never find out what particular zaak was excluded from the DL
I don't know if it's a necessary information or not and if it should be added to the audit trails

File: src/archiefvernietigingscomponent/destruction/management/commands/convert_log_templates.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+    def handle(self, *args, **options):
+        logs = TimelineLog.objects.all()
+
+        with transaction.atomic():
+            for log in logs:
+                template_name = log.template
+                if ".txt" in template_name:
+                    log.template = template_name.replace(".txt", ".html")
+                    log.save()
```
[the remark this answers is not in the document]
Comment (SilviaAmAm) [reply src/archiefvernietigingscomponent/destruction/management/commands/convert_log_templates.py:18]: This one-liner is beautiful

File: src/archiefvernietigingscomponent/destruction/templates/destruction/logs/item_destruction_failed.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+</code>
+
+<div class="log-item__body">
+    {% blocktrans with zaak=log.extra_data.zaak %}Case <strong>{{ zaak }}</strong> was not deleted.{% endblocktrans %}
+
+    <p class="log-item__details">
+        <code>
+            {{ log.extra_data.error|linebreaksbr|linenumbers }}
+        </code>
+    </p>
```
[the remark this answers is not in the document]
Comment (SilviaAmAm) [reply src/archiefvernietigingscomponent/destruction/templates/destruction/logs/item_destruction_failed.html:13]: Currently only the `aborted.html`, `created.html`, `review_created.html` and `updated.html` are used for the audit-trail report, but I renamed them all to `.html` to be consistent!

File: src/archiefvernietigingscomponent/destruction/templates/destruction/logs/aborted.html [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{% load i18n %}
+<code class="log-item__title" title="{{ log.timestamp.isoformat }}">
+    <strong>[{{ log.timestamp|date:'Y-m-d H:i' }}]:</strong>
```
[the remark this answers is not in the document]
Comment (SilviaAmAm) [reply src/archiefvernietigingscomponent/destruction/templates/destruction/logs/aborted.html:3]: I made an issue for this (https://github.com/maykinmedia/archiefvernietigingscomponent/issues/222): I need to figure out how to nicely handle CSS with weasyprint, for when the HTML is converted to a PDF.

**Comment by SilviaAmAm** [comment]
Latest version of the report :)
Screenshot from 2021-04-13 17-19-44

**Review by annashamray — APPROVED** [review]
Very nice!

**Pull request merged**